skills/bear-clean-style/scripts/scan_bear_clean_style.py

```
@@ -0,0 +1,671 @@
+    if not sel:
+        return None
+    i = sel.end()
+    depth = 0
+    n = len(text)
+    while i < n:
+        ch = text[i]
+        if ch == "(":
+            depth += 1
+        elif ch == ")":
+            depth -= 1
+        elif depth == 0 and text[i:i + 4].upper() == "FROM" and (i + 4 == n or not text[i + 4].isalnum() and text[i + 4] != "_"):
```

Ruff (0.15.14)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)